pkg/kubectl/history.go

```diff
@@ -131,6 +133,40 @@ func (h *DeploymentHistoryViewer) ViewHistory(namespace, name string, revision i
 	})
 }
 
+type StatefulSetHistoryViewer struct {
+	c clientset.Interface
+}
+
+// ViewHistory returns a list of the revision history of a statefulset
+// TODO: this should be a describer
+// TODO: needs to implement detailed revision view
+func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision int64) (string, error) {
+	sts, err := h.c.Apps().StatefulSets(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("failed to retrieve statefulset %s", err)
+	}
+	revisions, err := h.c.Apps().ControllerRevisions(namespace).List(metav1.ListOptions{LabelSelector: sts.Spec.Selector.String()})
```

Comment:
will this sometimes also report orphaned revisions ? To avoid that, we should further validate if the ownerReferences are for this set